R/derive_param_maxtemp.R

```diff
@@ -0,0 +1,136 @@
+#' Deriving maximum temperature record
+#'
+#' @description
+#' Create a summary record for maximum temperature per subject per period.
+#'
+#' @param dataset Input data set
+#'
+#'  Input data set should have the temperature record which will be derived by
+#'  using the function `derive_param_fever_occur`.
+#'
+#' @param filter_faobj filter for fever record
+#'
+#' *Default: "FEVER"*
+#' *Permitted Value: A Character Scalar*
+#'
+#' Helps to filter the fever record by passing the value from `FATESTCD` which
+#' indicates the fever records.
+#'
+#' @param by_vars Grouping variables
+#'
+#' *Default: vars(USUBJID,FAOBJ,ATPTREF)*
+#' *Permitted Value: A Character vector*
```

Comment:
We should switch to `exprs()` function as vars() is being superceded in dplyr - https://github.com/pharmaverse/admiral/issues/1627